extra_data/write_cxi.py

```diff
@@ -200,19 +220,22 @@ def write(self, filename):
                 dgrp['data_gain'] = h5py.SoftLink('/entry_1/data_gain')
 
             data = dgrp.create_virtual_dataset(
-                'data', layouts['data'], fillvalue=np.nan
+                'data', layouts['data'],
+                fillvalue=layouts['data'].dtype.type(_fillvalues['data'])
```

Comment:
Add a comment somewhere to say why we're doing this rather than just passing the number in? I think I recall that it was so that illegal values would error rather than being quietly converted to something else, but it would be good to record that in the code.

Also, if that is the rationale, does it make sense to do the check before we open the output file? Then if it fails, there's no incomplete file left over.

Reply:
Good point! I've moved it before file is created